synapse/handlers/appservice.py

```diff
@@ -100,10 +100,11 @@ async def handle_event(event):
 
                         if not self.started_scheduler:
 
-                            def start_scheduler():
-                                return self.scheduler.start().addErrback(
-                                    log_failure, "Application Services Failure"
-                                )
+                            async def start_scheduler():
+                                try:
+                                    return self.scheduler.start()
+                                except Exception:
+                                    log_failure("Application Services Failure")
```

Comment:
`log_failure` returns a function, but it doesn't look like that's then being passed to anything?

Reply:
Oh, I misinterpreted this. I should just change this into a `logger.error` call, I think.